internal/k8s/watch.go

```diff
@@ -48,39 +52,61 @@ func (kCli K8sClient) makeWatcher(f watcherFactory, ls labels.Selector) (watch.I
 }
 
 func (kCli K8sClient) WatchPods(ctx context.Context, ls labels.Selector) (<-chan *v1.Pod, error) {
-	ch := make(chan *v1.Pod)
-	watcher, err := kCli.makeWatcher(func(ns string) watcher {
+	// HACK(dmiller): There's no way to get errors out of an informer. See https://github.com/kubernetes/client-go/issues/155
+	// In the meantime, at least to get authorization and some other errors let's try to set up a watcher and then just
+	// throw it away.
+	_, err := kCli.makeWatcher(func(ns string) watcher {
 		return kCli.core.Pods(ns)
 	}, ls)
 	if err != nil {
-		return nil, errors.Wrap(err, "Pods.Watch")
+		return nil, errors.Wrap(err, "pods.Watch")
 	}
 
-	go func() {
-		for {
-			select {
-			case event, ok := <-watcher.ResultChan():
-				if !ok {
-					close(ch)
-					return
-				}
+	// HACK(dmiller): this can log to stderr. To prevent that we remove the error handler.
+	utilruntime.ErrorHandlers = []func(error){}
+	factory := informers.NewSharedInformerFactoryWithOptions(kCli.clientSet, 5*time.Second, informers.WithTweakListOptions(func(o *metav1.ListOptions) {
+		o.LabelSelector = ls.String()
+	}))
+	informer := factory.Core().V1().Pods().Informer()
 
-				if event.Object == nil {
-					continue
-				}
+	stopper := make(chan struct{})
+	ch := make(chan *v1.Pod)
 
-				pod, ok := event.Object.(*v1.Pod)
-				if !ok {
-					continue
-				}
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			mObj, ok := obj.(*v1.Pod)
+			if ok {
+				ch <- mObj
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
```

Comment:
this is ok for now, but i wonder if we should pass along that this was a Delete event. we have some logic in upper to try detect when the pod is being deleted but tbh it probably misses cases

Reply:
I agree that it would be great to do that in a future PR. I'll make a story for it.